stingray/crossspectrum.py

```diff
@@ -445,16 +452,17 @@ def _make_segment_spectrum(self, lc1, lc2, segment_size):
         if lc1.tseg != lc2.tseg:
             raise ValueError("Lightcurves do not have same tseg.")
 
-        # number of bins per segment
-        nbins = int(segment_size/lc1.dt)
-        start_ind = 0
-        end_ind = nbins
+        if gti is None:
+            gti = cross_two_gtis(lc1.gti, lc2.gti)
```

Comment:
I think I understand your point. I think I would solve this by asking the user to make a new Lightcurve object, but that's not all that important. 

But in the current implementation, they're *only* in `_make_segment_spectrum`, which is a helper function that user should never touch! Also, it looks like the user can *either* use the GTIs they put in, *or* those of the light curve. Is that the desired behaviour? If so, I think that should be documented somewhere. :)
Also (small nitpick), at the moment, the GTIs are listed as attributes, but I don't think they ever become attributes anywhere?